### PR TITLE
Reduce duplication in workflows

### LINKED ISSUE #678: Re-add a non-gpu dask client for GIF viz in small-scale docking when `LOCAL` dask client is overridden with a `LOCAL_GPU` client.
In small scale docking workflow when running MD using a LOCAL dask mode, we switch to a LOCAL_GPU mode when running MD as this correctly detects how many GPUs we have available as workers, rather than regular dask.

 However when we go to run gif viz we do not regenerate a CPU dask client, meaning we can only run as many workers as GPU, whereas GIF viz could be done a lot faster.

### PR DESCRIPTION
**Pull request by hmacdope**
## Description
Provide a brief description of the PR's purpose here.

Fixes #678 #606 #605 

Reduces duplication in creating dask clients and reading structures  /  ligands with meta factories. 

Also adds in a bunch of comments, TODOs and clarifications for workflows. 

## Developers certificate of origin
- [X] I certify that this contribution is covered by the MIT license as defined in our [LICENSE](https://github.com/choderalab/asapdiscovery/blob/main/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

### CODE REVIEW

**Comment by hmacdope** [comment]
@apayne97 @chrisiacovella @kaminow @ijpulidos this is probably ready for a first review from one or more of you. #701 will have to go in first though.

**Review by apayne97 — DISMISSED** [review]
I think I forgot to post this review, I don't think I had much substantial. I guess the thing I'd like to discuss tomorrow is whether we want to turn these workflows into classes or leave them as functions

File: asapdiscovery-data/asapdiscovery/data/schema_v2/meta_ligand_factory.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+from asapdiscovery.data.postera.postera_factory import PosteraFactory
+from asapdiscovery.data.schema_v2.ligand import Ligand
+from asapdiscovery.data.schema_v2.molfile import MolFileFactory
+from asapdiscovery.data.services_config import PosteraSettings
+from pydantic import BaseModel, Field, root_validator
+
+logger = logging.getLogger(__name__)
+
+
+class MetaLigandFactory(BaseModel):
```
[the remark this answers is not in the document]
Comment (hmacdope) [reply asapdiscovery-data/asapdiscovery/data/schema_v2/meta_ligand_factory.py:14]: yes, good spot.

File: asapdiscovery-data/asapdiscovery/data/schema_v2/meta_structure_factory.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+from asapdiscovery.data.schema_v2.complex import Complex
+from asapdiscovery.data.schema_v2.fragalysis import FragalysisFactory
+from asapdiscovery.data.schema_v2.structure_dir import StructureDirFactory
+from pydantic import BaseModel, Field, root_validator, validator
+
+logger = logging.getLogger(__name__)
+
+
+class MetaStructureFactory(BaseModel):
```
[the remark this answers is not in the document]
Comment (hmacdope) [reply asapdiscovery-data/asapdiscovery/data/schema_v2/meta_structure_factory.py:13]: yes, good spot.

**Comment by hmacdope** [comment]
@apayne97 @ijpulidos @chrisiacovella could I grab a review here?

**Comment by hmacdope** [comment]
@chrisiacovella @ijpulidos @apayne97 could I grab another review here?

**Review by chrisiacovella — APPROVED** [review]
I think this is probably fine to go.

File: asapdiscovery-docking/asapdiscovery/docking/workflows/large_scale_docking.py [hunk 3]
```diff
@@ -44,9 +40,10 @@
 from asapdiscovery.docking.workflows.workflows import PosteraDockingWorkflowInputs
 from asapdiscovery.ml.models import ASAPMLModelRegistry
 from asapdiscovery.modeling.protein_prep_v2 import ProteinPrepper
-from distributed import Client
 from pydantic import Field, PositiveInt, validator
 
+from asapdiscovery.data.dask_utils import
```
[the remark this answers is not in the document]
Comment (hmacdope) [reply asapdiscovery-docking/asapdiscovery/docking/workflows/large_scale_docking.py]: Ah thanks

**Pull request merged**